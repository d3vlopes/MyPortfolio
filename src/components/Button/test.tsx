import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Github } from '@styled-icons/boxicons-logos'

import Button from '.'

const { getByRole, getByTestId } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    renderWithTheme(
      <Button variant="primary" size="medium">
        Button
      </Button>,
    )

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#6D59A8',
      'font-size': '2.2rem',
      width: '21.5rem',
      height: '6.2rem',
    })
  })

  it('should render the variant secondary ', () => {
    renderWithTheme(<Button variant="secondary">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#83A848',
    })
  })

  it('should render the variant outline ', () => {
    renderWithTheme(<Button variant="outline">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      border: '1px solid #DEE7EA',
      'background-color': 'transparent',
    })
  })

  it('should render expand version', () => {
    renderWithTheme(<Button expand>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '31.5rem',
      height: '6.2rem',
    })
  })

  it('should render small size', () => {
    renderWithTheme(<Button size="small">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '1.8rem',
      width: '17.5rem',
      height: '5rem',
    })
  })

  it('should render large size', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '2.6rem',
      width: '28.8rem',
      height: '7.6rem',
    })
  })

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<Github data-testid="icon" />}>Button</Button>,
    )

    const button = getByRole('button', { name: /button/i })
    const icon = getByTestId('icon')

    expect(button).toBeInTheDocument()
    expect(icon).toBeInTheDocument()
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })

  it('should up while hover', async () => {
    renderWithTheme(<Button hoverUp>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyleRule('transform', 'translateY(-0.5rem)', {
      modifier: ':hover',
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Button>Button</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
