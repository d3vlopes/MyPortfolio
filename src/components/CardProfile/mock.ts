import { CardProfileProps } from '.'

export default {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul, RS',
  liking: 'Programação, design, música, séries e filmes.',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
} as CardProfileProps
