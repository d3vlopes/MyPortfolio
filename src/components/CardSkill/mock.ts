import { CardSkillProps } from '.'

export default [
  {
    icon: '/img/icons/sword.svg',
    title: 'Desafios',
    description:
      'Estou sempre em busca de novos desafios que me tirem do meu porto seguro.',
  },
  {
    icon: '/img/icons/target-arrow.svg',
    title: 'Persistência',
    description: `Não desanimo em frente aos obstáculos, quanto mais
      difícil, melhor a satisfação.`,
  },
  {
    icon: '/img/icons/lightbulb.svg',
    title: 'Agilidade',
    description: `Tenho facilidade para pensar em possíveis soluções para problemas.`,
  },
  {
    icon: '/img/icons/trending-up.svg',
    title: 'Evolução',
    description: `Gosto muito de aprender coisas novas e estar sempre sempre evoluindo.`,
  },
] as CardSkillProps[]
