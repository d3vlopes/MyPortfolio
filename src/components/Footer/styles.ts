import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: ${theme.spacings.xlarge} 0 ${theme.spacings.xsmall};
    background: ${theme.colors.bg.secondary.dark};

    @keyframes pulse {
      to {
        transform: scale(1.2);
      }
    }

    p {
      font-size: ${theme.font.sizes.small};
      line-height: 150%;
      color: ${theme.colors.grey};

      svg {
        color: red;

        &:hover {
          animation: pulse 0.25s infinite alternate;
        }
      }

      span:nth-child(2) {
        color: ${theme.colors.white};
      }
    }
  `}
`

export const ContactLinks = styled.div`
  ${({ theme }) => css`
    display: flex;
    gap: ${theme.spacings.xsmall};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`
