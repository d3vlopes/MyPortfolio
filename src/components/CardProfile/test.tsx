import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardProfile, { CardProfileProps } from '.'

const props: CardProfileProps = {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul - RS',
  liking: 'Programação, design, música, séries e filmes',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
}

const { getByRole, getByText } = screen

describe('<CardProfile />', () => {
  it('should render CardProfile', () => {
    const { container } = renderWithTheme(<CardProfile {...props} />)

    const image = getByRole('img', { name: 'Leandro Lopes' })
    const name = getByText('Leandro Lopes')
    const age = getByText('27 anos')
    const city = getByText('Santa Cruz do Sul - RS')
    const liking = getByText('Programação, design, música, séries e filmes')
    const dream = getByText(
      'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
    )

    expect(image).toHaveAttribute('src', '/img/me.jpg')
    expect(name).toBeInTheDocument()
    expect(age).toBeInTheDocument()
    expect(city).toBeInTheDocument()
    expect(liking).toBeInTheDocument()
    expect(dream).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
