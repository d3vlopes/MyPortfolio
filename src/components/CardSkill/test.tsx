import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Sword } from '@styled-icons/remix-fill/Sword'

import CardSkill from '.'
import mock from './mock'

const { getByTestId, getByRole, getByText } = screen

describe('<CardSkill />', () => {
  it('should render CardSkill', () => {
    renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    const icon = getByTestId('icon')
    const title = getByRole('heading', { name: /desafios/i })
    const description = getByText(
      /Estou sempre em busca de novos desafios que me tirem do meu porto seguro./i,
    )

    expect(icon).toBeInTheDocument()
    expect(title).toBeInTheDocument()
    expect(description).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
