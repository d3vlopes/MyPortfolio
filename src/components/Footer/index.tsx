import { Fragment } from 'react'
import { Heart } from '@styled-icons/boxicons-solid/Heart'

import ContactLink, { ContactLinkProps } from 'components/ContactLink'

import * as S from './styles'

export type FooterProps = {
  contactLinks: ContactLinkProps[]
}

const Footer = ({ contactLinks }: FooterProps) => {
  const renderContactLinks = (contact: ContactLinkProps) => {
    return (
      <Fragment key={`contact-${contact.name}`}>
        <ContactLink {...contact} />
      </Fragment>
    )
  }

  return (
    <S.Wrapper>
      <S.ContactLinks data-testid="contact-links">
        {contactLinks.map(renderContactLinks)}
      </S.ContactLinks>

      <p>
        Desenvolvido com <Heart aria-label="heart" size={25} /> por
        <span> Leandro Lopes</span>
      </p>
    </S.Wrapper>
  )
}

export default Footer
