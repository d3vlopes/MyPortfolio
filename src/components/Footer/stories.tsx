import { Story, Meta } from '@storybook/react'
import mock from 'components/ContactLink/mock'
import Footer, { FooterProps } from '.'

export default {
  title: 'Components/Footer',
  component: Footer,
  parameters: {
    layout: 'fullscreen',
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/KIQWZ47GDrrNCIxysbEqOA/My-Portfolio?node-id=900%3A1140',
    },
  },
  args: {
    contactLinks: mock,
  },
} as Meta<FooterProps>

export const Default: Story<FooterProps> = (args) => {
  return (
    <div>
      <Footer {...args} />
    </div>
  )
}
