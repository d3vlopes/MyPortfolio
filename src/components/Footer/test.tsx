import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import mock from 'components/ContactLink/mock'

import Footer, { FooterProps } from '.'
import { Props } from 'react-inlinesvg/esm'

const props: FooterProps = {
  contactLinks: mock,
}

const { getByLabelText, getByTestId } = screen

jest.mock('react-inlinesvg', () => {
  function ReactInlineSVG({ name }: Props) {
    return <svg aria-label={name} />
  }

  return ReactInlineSVG
})

describe('<Footer />', () => {
  it('should render Footer', () => {
    const { container } = renderWithTheme(<Footer {...props} />)

    const contactLinks = getByTestId('contact-links')
    const text = container.querySelector('p')
    const heartIcon = getByLabelText(/heart/i)

    expect(contactLinks).toBeInTheDocument()
    expect(text).toHaveTextContent(/Desenvolvido com por Leandro Lopes/i)
    expect(heartIcon).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
