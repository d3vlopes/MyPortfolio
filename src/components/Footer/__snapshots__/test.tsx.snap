// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render Footer 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4.8rem 0 1.6rem;
  background: #302F43;
}

.c0 p {
  font-size: 1.8rem;
  line-height: 150%;
  color: rgba(255,255,255,0.5);
}

.c0 p svg {
  color: red;
}

.c0 p svg:hover {
  -webkit-animation: pulse 0.25s infinite alternate;
  animation: pulse 0.25s infinite alternate;
}

.c0 p span:nth-child(2) {
  color: #FFFFFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
  margin-bottom: 1.6rem;
}

@media (max-width:1170px) {

}

<div
  class="c0"
>
  <div
    class="c1"
    data-testid="contact-links"
  >
    <div
      class=""
    >
      <a
        href="https://github.com/d3vlopes"
        rel="noopenner, noreferrer"
        target="_blank"
      >
        <svg
          aria-label="Github"
        />
      </a>
    </div>
    <div
      class=""
    >
      <a
        href="https://www.linkedin.com/in/d3vlopes/"
        rel="noopenner, noreferrer"
        target="_blank"
      >
        <svg
          aria-label="Linkedin"
        />
      </a>
    </div>
    <div
      class=""
    >
      <a
        href="https://api.whatsapp.com/send?phone=5551997465898&text=Contato%20pelo%20site"
        rel="noopenner, noreferrer"
        target="_blank"
      >
        <svg
          aria-label="Whatsapp"
        />
      </a>
    </div>
  </div>
  <p>
    Desenvolvido com 
    <svg
      aria-hidden="true"
      aria-label="heart"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="25"
      viewBox="0 0 24 24"
      width="25"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.205 4.791a5.938 5.938 0 0 0-4.209-1.754A5.906 5.906 0 0 0 12 4.595a5.904 5.904 0 0 0-3.996-1.558 5.942 5.942 0 0 0-4.213 1.758c-2.353 2.363-2.352 6.059.002 8.412L12 21.414l8.207-8.207c2.354-2.353 2.355-6.049-.002-8.416z"
      />
    </svg>
     por
    <span>
       Leandro Lopes
    </span>
  </p>
</div>
`
